### ISSUE TITLE
Multimedia player - play button isn't tagged as play

### ISSUE THREAD

**Issue by neoinsight** [wet-boew/wet-boew#7955]
Screen reader participants had a hard time figuring out how to play the video - the play button isn't labelled as Play. Instead it reads out only the video title - in this case "Check out this incredible footage.."   They expected it to say play - like Youtube does. 

The multimedia player we tested is posted on this page http://www.forces.gc.ca/en/news/article.page?doc=check-out-this-incredible-footage-of-the-evacuation/iw7nrjtp

Here is a link to the spot in the video for a participant on Voice Over: https://youtu.be/-68k_Wwby7E?t=26m01s  (wind back further and you'll see that at first she only finds the transcript, takes a while to find the video because of the missing play control)
Another participant on NVDA: https://youtu.be/nwPUfzD1yh0?t=25m56s

**Comment by LaurentGoderre** [comment]
@neoinsight the play button is labelled, this is a bug between screen readers and the video element. WE encountered this at Statcan as well.

**Comment by neoinsight** [comment]
@LaurentGoderre So this isn't a WET issue? Because they hadn't encountered this before with other video players.  A third participant just emailed to say the same:

> "And I was able to play the video, though that kind of control is frustrating because there is no way to really know if something will happen if you spacebar on the text if it just says “Video: Check out this incredible footage of the evacuation.” I wonder what setting was messed u

p?"

**Comment by kodecount** [comment]
Just a quick look at the code could it be `aria-labelledby="wb-auto-1-md"` this line of code conflicting?

**Comment by kodecount** [comment]
Do we need both `aria-controls="wb-auto-1-md"` and ` aria-labelledby="wb-auto-1-md" `?

**Comment by LaurentGoderre** [comment]
@neoinsight yes, the issue is not thast it isn't labelled it that the label aren't read out. We discovered that whitespace in the video element seemed to matter for some odd reason.

The aria attribute is also why the video title is read for each control but that was requested by the AWG

/cc @SBleeks @dfarough

**Comment by neoinsight** [comment]
Since I have some experts here - it also looks like the page didn't code the transcript properly either- it's not associated with the video at all, so some of the participants on screen-readers didn't find it until I asked them to do so.

**Comment by neoinsight** [comment]
@kodecount Is this something the person who coded the page should resolve, rather than a WET issue? I'll close the issue if that is the case.

**Comment by kodecount** [comment]
No/Yes the issue is complicated. Right now we use `aria-controls` and `aria-labelledby` which are added by WET. @LaurentGoderre said these were added to fix an issue with a specific screen reader it seems, but some screen readers are reading the video title due to `aria-labelledby` and it only in some browsers from my quick testing.

**Comment by dorina1** [comment]
@LaurentGoderre; @wet-boew/access-working-group  My issue with the multimedia player [Volume Toggle button](https://github.com/wet-boew/wet-boew/issues/8030) had nothing to do with the play/pause button issue. 

JAWS 18 can handle very well many ARIA attributes but I cannot hear the proper message when my focus is on the volume slider. 

![capture](https://cloud.githubusercontent.com/assets/1918938/26529878/f7fdfdb2-4396-11e7-9ba7-96384745d6e1.PNG)

Could you please point me to where exactly is the ARIA message coded (in which file?) because I would like to confirm that the message is coded indeed like this (or similar but to indicate the full functionality):
**"Volume left right slider (number will indicate volume level). To increase or decrease use the up/down arrow keys. Media left right slider (number will indicate video/audio position).  To forward or backward the media file use the right/left arrow keys. ".** 

This is the way the slider works for keyboard access and for mouse users, however to a JAWS user all the information that is currently given is : "Volume left right slider (number will indicate volume level) to increase or decrease use the arrow keys".

We need to know if the ARIA message is indeed the issue or if JAWS is the issue so we could report it to FS.

Could you please point to the file where this ARIA message is coded? Thank you!

**Comment by LaurentGoderre** [comment]
https://github.com/wet-boew/wet-boew/blob/master/src/plugins/multimedia/assets/mediacontrols.html

**Comment by dorina1** [comment]
@LaurentGoderre  Thanks for sending me the link.  I cannot figure it out from where exactly we hear this message "Volume left right slider (number will indicate volume level) to increase or decrease use the arrow keys" when the Volume slider is in focus, but the verbal message does not properly describe the functionality of the volume slider, so the it is not accessible for JAWS users.

**Comment by LaurentGoderre** [comment]
how doesn't it describe it? Tat's what it does no?

**Comment by dorina1** [comment]
The current description when navigated with JAWS is:
"Volume left right slider 75 (number indicates volume level) to increase or decrease use the arrow keys"

But the functionality of it is different when navigated with JAWS or with keyboard access only:
- if we use the up/down arrow keys, it will increase/decrease the volume
- if we use right/left arrow keys , it will forward/backward the media (audio or video) file

**Comment by LaurentGoderre** [comment]
Hmmm...it's the left right slider......I think JAWS reports that. 

It's annoying because the way it's design, form anywhere on the player, even when not in the controls, you can change the voume using up and down and sccroll the video using the left and right arrow. We might need to change the behavior to making the left and right key change the volume when the focus is on the slider but revert to scrolling when focus has moved on.

**Comment by dorina1** [comment]
Yes, you are correct, for keyboard-only users it is true, they can change the volume or scroll the video form anywhere in the player, but this is not the case for JAWS users.

JAWS users must be focused on the left right slider to be able to change the volume or media file position. 

I agree with your recommended approach: to keep the volume slider for volume and the audio/video slider for the media file. Currently they are somehow linked and not separate when navigated with JAWS 18.

![multimedia-player](https://cloud.githubusercontent.com/assets/1918938/26553557/26aff39a-445a-11e7-886d-7fb44bd9d41b.jpg)

**Comment by LaurentGoderre** [comment]
Seriously?? up and down arrow don't work in JAWS??

**Comment by dorina1** [comment]
They work as I mentioned before: **JAWS users must be focused on the left right slider to be able to change the volume or media file position.**

**Comment by dorina1** [comment]
what do you mean? JAWS does not ignore arrows, we can use the arrows up/down or left/right but we must have the focus on that specific slider in order to play with the volume and the media location. 

Keyboard-only users don't necessary need to be focused on the volume slider, like you said , they can play with the volume and media location from anywhere in the multimedia container.

**Comment by LaurentGoderre** [comment]
Why do JAWS need to focus on the element? WHy doesn't it work if they try to use the up/down arrow when not focus on the slider?

**Comment by SBleeks** [comment]
Focus needs to be placed within the player/ on the slider. Then turn forms mode off Insert Z, Now you can press spacebar to pause/play, left arrow to rewind, right arrow to fast forward, up arrow to increase volume, down arrow to decrease volume.

Please note: the video used as an example for the player is terrible. It’s an armed forces video with no speech. It’s only sounds like a vacuum cleaner or maybe a jet taking off or maybe someone using a blow dryer…

**Comment by LaurentGoderre** [comment]
Is there a way to avoid having to use screen moe? Using role=application?

**Comment by SBleeks** [comment]
Forms mode must be off in order to interact with the player.  (don’t hate the player, hate da game)

**Comment by dfarough** [comment]
I suggest that you read the following article which does a very good job of explaining the screen reader interaction modes.  
https://tink.uk/understanding-screen-reader-interaction-modes/

The screen reader needs to have the proper context in order to determine what to do when a particular key is pressed.  When you turn the virtual cursor off, the browser will then be able to react to the keystrokes that Jaws would normally intercept.

**Comment by nricherbrule** [comment]
I believe this issue is resolved @bsouster

**Comment by Bhawna06** [comment]
As this issue is resolved, in my opinion too... Shouldn't it be closed??

**Comment by dfarough** [comment]
@wet-boew/access-working-group 
[@LaurentGoderre asked why a screen reader would not honor the arrow keys  while in the media player.
Check out the following article which explains the interaction modes of screen readers.
(https://tink.uk/understanding-screen-reader-interaction-modes/

Care should be taken not to use the application role inappropriately because it will disable the browse mode and make it difficult for screen reader users to interact with the page.

The play button is now being spoken correctly using Jaws. the volume button when focused does work as expected, but the changed values are not being reported by Jaws.  The values for the progress bar are not being spoken when the left and right arrow keys are used to go forward and backward through the video.  This is also true when using NVDA.
@SBleeks is correct about turning off browse mode .  If a user knows enough to do this, the volume and video slider work as expected when the arrow keys are pressed and the page works exactly as it does for keyboard only users.
Using NVDA, you can turn off browse mode by pressing the NVDA key (Caps LOck) and the spacebar.
@Bhawna06 this could be closed I think, but I am not sure about the benefit being provided by the aria-controls attribute on the video content.   It is my understanding that Jaws is the only screen reader that supports this and I believe that the support has changed in the latest version.
It seems to me that tabbing through the elements is much more verbose than necessary.
When tabbing through the controls for the video player, Jaws provides the following output
Play
video video Toggle button Use JawsKey+Alt+M to move to controlled element 
Mute
video video left right slider Use JawsKey+Alt+M to move to controlled element 
75
Volume
Progress bar partially checked Read Only 
video video left right slider Use JawsKey+Alt+M to move to controlled element 
75
Volume
